optaplanner-core/src/main/java/org/optaplanner/core/api/score/constraint/ConstraintMatch.java

```diff
@@ -91,9 +100,10 @@ public int compareTo(ConstraintMatch<Score_> other) {
             /*
              * TODO Come up with a better cache.
              *
-             * We reuse the comparator from here, since it internally caches some reflection that we don't want to be performing
-             * over and over again. However, there are possibly thousands of instances of this class, and each will get its own
-             * comparator. Therefore, the caching is only partially effective.
+             * We reuse the comparator from here, since it internally caches some reflection that we don't want to be
```

Comment:
Avoid "we" and "that we don't want to be ...ing".
```suggestion
             * Reuse the comparator to internally caches reflection for performance benefits.
```